activerecord/lib/active_record/migration.rb

```diff
@@ -1374,9 +1374,14 @@ def use_advisory_lock?
         Base.connection.advisory_locks_enabled?
       end
 
+      AdvisoryLockBase = Class.new(ActiveRecord::Base) do
+        self.connection_specification_name = "AdvisoryLockBase"
+      end
```

Comment:
For `SchemaMigration` and `InternalMetadata` we do

``` ruby
def self._internal?
  true
end
```

Feels like we should do the same here

Reply:
Also maybe this should be `self.abstract_class = true`?